src/syscall/syscall_libc_darwin.go

```diff
@@ -188,6 +198,10 @@ func Lstat(path string, p *Stat_t) (err error) {
 //export stat$INODE64
 func libc_stat(pathname *byte, ptr unsafe.Pointer) int32
 
+// int fstat(int fd, struct stat * buf);
+//export fstat$INODE64
+func libc_fstat(fd int, ptr unsafe.Pointer) int32
```

Comment:
Note that a C int us different from a Go int. On a 64 bit architecture, C usually uses a 32 bit int while Go usually uses a 64-bit int. Use int32 to make sure it's always a 32-bit int.

Reply:
Hmm... so what should the type of `fd` be here then?

Reply:
`int32`, to match C. Otherwise you might run into weird ABI issues.